src/app/CMakeLists.txt

```diff
@@ -77,31 +77,95 @@ set_source_files_properties(
 if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
     # substitute @EXECUTABLE@ in dist/mac/Info.plist
     get_target_property(EXECUTABLE qbt_app OUTPUT_NAME)
-    configure_file(${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist
-        ${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist @ONLY)
+    configure_file("${qBittorrent_SOURCE_DIR}/dist/mac/Info.plist"
+        "${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist" @ONLY)
     file(GENERATE
-        OUTPUT ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
-        INPUT ${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist
+        OUTPUT "${qBittorrent_BINARY_DIR}/dist/mac/Info.plist"
+        INPUT "${qBittorrent_BINARY_DIR}/dist/mac/pregen.plist"
     )
     set_target_properties(qbt_app PROPERTIES
         MACOSX_BUNDLE ON
         MACOSX_BUNDLE_BUNDLE_NAME "${EXECUTABLE}"
-        MACOSX_BUNDLE_INFO_PLIST ${qBittorrent_BINARY_DIR}/dist/mac/Info.plist
+        MACOSX_BUNDLE_INFO_PLIST "${qBittorrent_BINARY_DIR}/dist/mac/Info.plist"
     )
     target_sources(qbt_app PRIVATE
-        ${QT_TRANSLATIONS}
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qt.conf
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qBitTorrentDocument.icns
-        ${qBittorrent_SOURCE_DIR}/dist/mac/qbittorrent_mac.icns
+        ${QT_TRANSLATIONS_LIST}
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qt.conf"
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qBitTorrentDocument.icns"
+        "${qBittorrent_SOURCE_DIR}/dist/mac/qbittorrent_mac.icns"
     )
 elseif (CMAKE_SYSTEM_NAME STREQUAL "Windows")
     set_target_properties(qbt_app PROPERTIES WIN32_EXECUTABLE ON)
     if (MINGW)
-        target_sources(qbt_app PRIVATE ${qBittorrent_SOURCE_DIR}/src/qbittorrent_mingw.rc)
+        target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent_mingw.rc")
     else()
-        target_sources(qbt_app PRIVATE ${qBittorrent_SOURCE_DIR}/src/qbittorrent.rc)
+        target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent.rc")
+    endif()
+    target_sources(qbt_app PRIVATE "${qBittorrent_SOURCE_DIR}/src/qbittorrent.exe.manifest")
+
+    # Copy DLLs to the build folder after the build, so that the executable can be run from there.
+    # When using vcpkg, everything is automatically copied except for some stuff handled by windeployqt;
+    # this is no problem because the copy_if_different command is idempotent in this case
+    # TODO: works for MSVC, what about MinGW?
+    # TODO: This can probably be greatly simplified by using TARGET_RUNTIME_DLLS instead, in CMake >= 3.21
+    if (MSVC_RUNTIME_DYNAMIC)
+        # Must manually find these DLLs, because the target names for these libs point at the .lib files, not the .dll files
+        # Handling zlib is trickier than OpenSSL because FindZLIB sucks
+        cmake_path(SET OPENSSL_DLLS_PATH "${OPENSSL_CRYPTO_LIBRARY}")
+        cmake_path(GET OPENSSL_DLLS_PATH PARENT_PATH OPENSSL_DLLS_PATH)
```

Comment:
I'm pretty sure CMake knows the exact name of the \.dll file, so you can avoid this kind of guessing.